### ISSUE TITLE
Creating a project with more than 99 gems enabled throws an error and prevents gems configuration

### ISSUE THREAD

**Issue by LB-KacperBieniek** [o3de/o3de#15136]
**Describe the bug**
When the user either creates a new project or edits an existing one and tries to enable 100 gems (or more), the Project Manager error prompt appears:

```
Cannot add gem Script Canvas Developer to project.

Error:
ResolutionTooDeep: 100 At: C:\o3de-install\python\runtime\python-3.10.5-rev1-windows\python\Lib\site-packages\resolvelib\resolvers.py(432): resolve C:\o3de-install\python\runtime\python-3.10.5-rev1-windows\python\Lib\site-packages\resolvelib\resolvers.py(521): resolve c:\o3de-install\scripts\o3de\o3de\compatibility.py(533): resolve_gem_dependencies c:\o3de-install\scripts\o3de\o3de\compatibility.py(198): get_gem_project_incompatibile_objects c:\o3de-install\scripts\o3de\o3de\enable_gem.py(96): enable_gem_in_project

```

Followed by another prompt:

`Failed to configure gems for template.`

The Script Canvas Developer gem is the first gem to throw the prompt when using the Default project template and enabling all inactive gems, but this issue applies to all gems available in project configuration. Whichever gem is enabled as the 100th one it will cause the error to appear.

The issue occurs on Stabilization/2305 ([9e5047f](https://github.com/o3de/o3de/commit/9e5047fef0a61cfb3044342f0f1fd468366dcf7f)), Development ([d6ef267](https://github.com/o3de/o3de/commit/d6ef267abd8a427dcfacfd82b12678a4a4581409)), and on the latest Development installer. It also occurs on Windows and Linux for both Source and Pre-built configurations.

**Steps to reproduce**
1. Launch the Project Manager.
2. Create a new project.
3. Choose the Default template project.
4. Configure the project with more gems.
5. Enable all Inactive gems.
6. Press the _Create Project_ button.

**Expected behavior**
The project is created and ready to be built.

**Actual behavior**
The Project Manager prompts the user with an error about a failed attempt to add a gem to the project.

**Screenshots/Video**

![image](https://user-images.githubusercontent.com/87059746/225607899-c2670c96-04f8-4bbf-a57a-1dcaa1b524db.png)
![image](https://user-images.githubusercontent.com/87059746/225607926-5c527a27-415c-4299-b143-c2e640972840.png)

**Found in Branch**
Stabilization/2305

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
[9e5047f](https://github.com/o3de/o3de/commit/9e5047fef0a61cfb3044342f0f1fd468366dcf7f)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: AMD Ryzen 7 3800X 8-Core Processor
 - GPU: AMD Radeon RX 5700 XT
 - Memory: 16GB

**Comment by AMZN-alexpete** [comment]
This fix has been merged to stabilization and development

**Comment by LB-KacperBieniek** [comment]
Issue verified as fixed on latest Stabilization/2305 (602a599) and Development (beb8bce).
